### PR TITLE
Parse PyPI URL from GitHub issue for RC testing

### PR DESCRIPTION
**Pull request by Lee-W**
* ci(dev/scripts): parse pinned_package from airflow package release testing announcement github issue
* ci(dev/scripts): add requirements.txt for pinning used package versions
* ci(Makefile): update test-rc-deps for dev/scripts/replace_dependencies.py new feature

#901

### CODE REVIEW

File: dev/scripts/replace_dependencies.py [hunk 1]
```diff
@@ -3,10 +3,76 @@
...
+    download_package_url = f"{cncf_url}#files"
+    req = requests.get(download_package_url)
+    soup = BeautifulSoup(req.text, features="html5lib")
+    first_file_card_comment = soup.find("div", {"class": "file__card"})
+    url = first_file_card_comment.a.get("href")
+    return f"apache-airflow-providers-cncf-kubernetes @{url}"
+
+
+def _parse_pinned_package_from_pypi_url(url: str) -> str:
+    """Parase the pinned version from PyPI URL
```
Comment (phanikumv) [inline dev/scripts/replace_dependencies.py:36]:
```suggestion
    """Parse the pinned version from PyPI URL
```

Comment (Lee-W) [reply dev/scripts/replace_dependencies.py]: Thanks! just fix the typo

**Review by phanikumv — COMMENTED** [review]
can we name the folder as `dev/providers_rc_test_scripts` ?

**Review by pankajkoti — COMMENTED** [review]
would be nice to see a quick demo of this :-) Also, a good topic for our R&D demos later.

**Comment by Lee-W** [comment]
> can we name the folder as `dev/providers_rc_test_scripts` ?

Sure! Just renamed it

**Review by phanikumv — COMMENTED** [review]
we now have a `replace_dependencies.py` in both providers_rc_test_scripts and scripts folder. is this intended?

**Comment by Lee-W** [comment]
> we now have a `replace_dependencies.py` in both providers_rc_test_scripts and scripts folder. is this intended?

@phanikumv  I cannot find another `replace_dependencies.py`. Could you please share the commit sha you're on and the full path of both `replace_dependencies.py` so that I can check again? Thanks!

**Comment by pankajastro** [comment]
> > we now have a `replace_dependencies.py` in both providers_rc_test_scripts and scripts folder. is this intended?
> 
> @phanikumv I cannot find another `replace_dependencies.py`. Could you please share the commit sha you're on and the full path of both `replace_dependencies.py` so that I can check again? Thanks!

look like `replace_dependencies.py` has been moved from scripts to providers_rc_test_scripts

**Review by pankajastro — APPROVED** [review]
Nice! now I think we can also update the CI job to accept the GH issue and use that to build an image and update the deployment so that the local machine would be free.

**Comment by Lee-W** [comment]
> I think we can also update the CI job to accept the GH issue and use that to build an image and update the deployment 

Hi @pankajastro , thanks for your suggestion, but I'm a bit confused on this one. Could you please describe a bit more? Thanks!

**Comment by pankajastro** [comment]
> Hi @pankajastro , thanks for your suggestion, but I'm a bit confused on this one. Could you please describe a bit more? Thanks!

I was thinking of a CI job which takes airflow provider release testing GitHub issues as input  and updated the deployment  based on the provider version on that issue

**Pull request merged**